packages/teuchos/numerics/src/Teuchos_SerialSymDenseMatrix.hpp

```diff
@@ -631,6 +637,29 @@ int SerialSymDenseMatrix<OrdinalType, ScalarType>::putScalar( const ScalarType v
   return 0;
 }
 
+template<typename OrdinalType, typename ScalarType> void
+SerialSymDenseMatrix<OrdinalType, ScalarType>::swap(
+  SerialSymDenseMatrix<OrdinalType, ScalarType> &B)
+{
+  // cache B values
```

Comment:
Just use std::swap, instead of making temporaries.

Reply:
I just brought in Mike's changes and used the std utility library.  I verified that both Dakota and Teuchos unit tests work with these changes.